spring-vault-core/src/main/java/org/springframework/vault/core/VaultKeyValue2Template.java

```diff
@@ -89,4 +90,29 @@ public void put(String path, Object body) {
 		doWrite(createDataPath(path), Collections.singletonMap("data", body));
 	}
 
+	/**
+	 * Performs a KV Patch operation.
+	 * @param path must not be {@literal null} or empty.
+	 * @param kv New key value map to be updated
+	 * @since 2.3
+	 */
+	public void patch(String path, Map<String, ?> kv) {
+		Assert.hasText(path, "Path must not be empty");
+
+		// To do patch operation, we need to do a read operation first
+		VaultResponse readResponse = get(path);
+		Assert.notNull(readResponse, "VaultResponse must not be null");
+		Assert.notNull(readResponse.getData(),
+				String.format("No data found at %s; patch only works on existing data", path));
+		Assert.notNull(readResponse.getMetadata(), "Metadata must not be null");
+		Map<String, Object> data = readResponse.getData();
+		Map<String, Object> metadata = readResponse.getMetadata();
+		kv.forEach(data::put);
+		Map<String, Object> body = new HashMap<>();
+		body.put("data", data);
+		body.put("options", Collections.singletonMap("cas", metadata.get("version")));
+
+		doWrite(createDataPath(path), body);
```

Comment:
There should be also a reaction on whether the write was successful. A concurrent process could have updated the secret in the meantime. Returning `true` on success (similar to Java's `compareAndSet` methods in the concurrency util package) would make sense.

Reply:
I changed the return type as boolean and true if the response is successful, but please let me know if you want another return type such as VaultResponse itself or something else:
https://github.com/spring-projects/spring-vault/pull/587/commits/95d25bd9120c340cecc2eab71c67ffc047e41f07#diff-917f198074796659086e92ad34820ddfR120-R125

And FYI, according to the document of HashCorp's KV secret engine V2, we will be safe from the concurrency issue because Vault will not allow if the secret's versioning is not correct.(If the existing version of the secret is 3 and incoming one is still 3 it won't allow writing operation so it can avoid from the race condition)
Please see the following links:
https://learn.hashicorp.com/tutorials/vault/versioned-kv
https://www.vaultproject.io/docs/commands/kv/patch